internal/discovery/dependencydiscovery.go

```diff
@@ -290,8 +290,21 @@ func (dd *DependencyDiscovery) dependencyToDiscover(
 		// Record in report as excluded external dependency
 		if dd.report != nil {
 			absPath := util.CleanPath(depPath)
-			run, _ := dd.report.EnsureRun(absPath)
-			_ = dd.report.EndRun(run.Path, report.WithResult(report.ResultExcluded), report.WithReason(report.ReasonExcludeExternal))
+
+			run, err := dd.report.EnsureRun(l, absPath)
+			if err != nil {
+				l.Errorf("Error ensuring run for excluded external dependency %s: %v", absPath, err)
+			}
+
+			err = dd.report.EndRun(
+				l,
+				run.Path,
```

Comment:
`run` can be nil? if `run, err := dd.report.EnsureRun()` will return an error?

Can be used `depPath`?